spaceship/lib/spaceship/tunes/availability.rb

```diff
@@ -106,9 +104,20 @@ def enable_b2b_app!
       # Adds users for b2b enabled apps
       def add_b2b_users(user_list = [])
         raise "Cannot add b2b users if b2b is not enabled" unless b2b_app_enabled
-        @b2b_users = user_list.map do |user|
-          B2bUser.from_username(user)
-        end
+        @b2b_users = user_list.map { |user| B2bUser.from_username(user) }
+        self
+      end
+
+      # Updates users for b2b enabled apps
+      def update_b2b_users(user_list = [])
+        raise "Cannot add b2b users if b2b is not enabled" unless b2b_app_enabled
+        added_users = b2b_users.map(&:ds_username)
+        return self if (added_users - user_list) == (user_list - added_users)
+        users_to_add = user_list.reject { |user| added_users.include?(user) }
+        users_to_remove = added_users.reject { |user| user_list.include?(user) }
+        @b2b_users = b2b_users.reject { |user| users_to_remove.include?(user.ds_username) }
+        @b2b_users.concat(users_to_add.map { |user| B2bUser.from_username(user) })
+        @b2b_users.concat(users_to_remove.map { |user| B2bUser.from_username(user, is_add_type: false) })
         self
```

Comment:
Same here. Can we change this to `return self`?

Reply:
Can we also had some space between these lines with some comments so its a bit easier on the eyes? 😇 Something like...

```rb
      # Updates users for b2b enabled apps
      def update_b2b_users(user_list = [])
        raise "Cannot add b2b users if b2b is not enabled" unless b2b_app_enabled

        # Returns if list is unchanged
        added_users = b2b_users.map(&:ds_username)
        return self if (added_users - user_list) == (user_list - added_users)

        users_to_add = user_list.reject { |user| added_users.include?(user) }
        users_to_remove = added_users.reject { |user| user_list.include?(user) }

        @b2b_users = b2b_users.reject { |user| users_to_remove.include?(user.ds_username) }
        @b2b_users.concat(users_to_add.map { |user| B2bUser.from_username(user) })
        @b2b_users.concat(users_to_remove.map { |user| B2bUser.from_username(user, is_add_type: false) })

        return self
end
```

Reply:
Sure thing. I will add comments elsewhere as well. : )